vispy/visuals/volume.py

```diff
@@ -458,8 +491,16 @@ def __init__(self, vol, clim=None, method='mip', threshold=None,
             ], dtype=np.float32))
         
         self._interpolation = interpolation
-        self._tex = tex_cls((10, 10, 10), interpolation=self._interpolation, 
-                            wrapping='clamp_to_edge')
+        tex_kwargs = {}
+        if texture_format and not isinstance(texture_format, str):
+            if texture_format not in self._texture_dtype_format:
+                raise ValueError("Can't determine internal texture format for '{}'".format(texture_format))
+            texture_format = self._texture_dtype_format[texture_format]
+        if isinstance(texture_format, str):
+            tex_kwargs['internalformat'] = texture_format
+            tex_kwargs['format'] = 'luminance'
+        self._tex = tex_cls((10, 10, 10), interpolation=self._interpolation,
+                            wrapping='clamp_to_edge', **tex_kwargs)
```

Comment:
Reading this as a naive newcomer, this line is very mysterious. What is (10, 10, 10)? Why clamp to edge?

Reply:
I'll add some comments. I think the 10 is just a placeholder size so that the texture is created. It will get resized when actual data is uploaded. The clamp_to_edge...I don't remember.

Reply:
Separate the texture creation out and added some comments so hopefully this is a little more clear.

Reply:
Awesome, thank you!